lib/fdo.c

```diff
@@ -843,151 +854,87 @@ static bool _STATE_DI(void)
-			is_mfg_addr = true;
+		} else {
+			LOG(LOG_DEBUG, "Manufacturer Transport protocol: HTTP\n");
 		}
-	}
-	if (is_mfg_addr == false) {
-		LOG(LOG_ERROR, "Failed to get neither ip/dn mfg address\n");
-		ERROR()
-		goto end;
-	}
-#else
-#ifdef MANUFACTURER_IP
-	manIPAddr = fdo_ipaddress_alloc();
-	if (!manIPAddr) {
-		LOG(LOG_ERROR, "Failed to alloc memory\n");
-		ERROR()
-		goto end;
-	}
-	int result = fdo_printable_to_net(MANUFACTURER_IP, manIPAddr->addr);
-
-	if (result <= 0) {
-		LOG(LOG_ERROR, "Failed to convert Mfg address\n");
-		ERROR()
-		goto end;
-	}
-	manIPAddr->length = IPV4_ADDR_LEN;
-#endif
-
-	const char *mfg_dns = NULL;
-#ifdef MANUFACTURER_DN
-	mfg_dns = MANUFACTURER_DN;
-#endif
-#endif
 
-	/* If MANUFACTURER_PORT file does not exist or is a blank file then,
-	 *  use existing global DI port(8039) else use configured value as DI
-	 *  port
-	 */
-
-	fsize = fdo_blob_size((char *)MANUFACTURER_PORT, FDO_SDK_RAW_DATA);
-
-	if ((fsize > 0) && (fsize <= FDO_PORT_MAX_LEN)) {
-		char port_buffer[FDO_PORT_MAX_LEN + 1] = {0};
-		char *extra_string = NULL;
-		unsigned long configured_port = 0;
-
-		if (fdo_blob_read((char *)MANUFACTURER_PORT, FDO_SDK_RAW_DATA,
-				  (uint8_t *)port_buffer, fsize) == -1) {
-			LOG(LOG_ERROR, "Failed to read manufacturer port\n");
+		// parse IP/DNS. check for IP first, then DNS
+		mfg_ip = fdo_ipaddress_alloc();
+		if (!mfg_ip) {
+			LOG(LOG_ERROR, "Failed to alloc memory\n");
+			ERROR();
 			goto end;
 		}
-
-		configured_port = strtoul(port_buffer, &extra_string, 10);
-
-		if (strnlen_s(extra_string, 1)) {
-			LOG(LOG_ERROR, "Invalid character encounered in the "
-				       "given port.\n");
-			goto end;
+		result = fdo_printable_to_net(mfg_dns, mfg_ip->addr);
+		if (result > 0) {
+			// valid IP address
+			mfg_ip->length = IPV4_ADDR_LEN;
+			LOG(LOG_DEBUG, "Manufacturer IP will be used\n");
+		} else if (result == 0) {
+			// not an IP address, so treat it as DNS address
+			LOG(LOG_DEBUG, "Manufacturer DNS will be used\n");
 		}
 
-		if (!((configured_port >= FDO_PORT_MIN_VALUE) &&
-		      (configured_port <= FDO_PORT_MAX_VALUE))) {
+		// parse port
+		if (!((di_port >= FDO_PORT_MIN_VALUE) &&
+		      (di_port <= FDO_PORT_MAX_VALUE))) {
 			LOG(LOG_ERROR,
 			    "Manufacturer port value should be between "
 			    "[%d-%d].\n",
 			    FDO_PORT_MIN_VALUE, FDO_PORT_MAX_VALUE);
-			goto end;
+			// set default
+			di_port = g_DI_PORT;
 		}
-
-		di_port = (uint16_t)configured_port;
-
-	} else if (fsize > FDO_PORT_MAX_LEN) {
-		LOG(LOG_ERROR,
-		    "Manufacturer port value should be between "
-		    "[%d-%d]. "
-		    "It should not be zero prepended.\n",
-		    FDO_PORT_MIN_VALUE, FDO_PORT_MAX_VALUE);
-		goto end;
+		LOG(LOG_DEBUG, "Manufacturer Port: %d\n", di_port);
 	}
 
-	LOG(LOG_DEBUG, "Manufacturer Port = %d.\n", di_port);
-
 	prot_ctx = fdo_prot_ctx_alloc(fdo_process_states, &g_fdo_data->prot,
-				      manIPAddr, mfg_dns, di_port, false);
+				      mfg_ip, mfg_dns, di_port, tls);
```

Comment:
If we don't pass a NULL mfg_dns, the mfg_ip is not considered while running the fdo_prot_ctx_run.

If we wish to use mfg_ip, we should create another pointer that should either be NULL when mfg_ip is valid and then pass that to fdo_prot_ctx_alloc().

Based on current observation, passing an IP as DNS is okay as the internal network library can perform network resolution. If that is the case, we can remove the mfg_ip altogether.